rom/laghos.cpp

```diff
@@ -1292,27 +1405,41 @@ int main(int argc, char *argv[])
                 // TODO: it needs to be save in the format of HDF5 format
                 // TODO: how about parallel version? introduce rank in filename
                 // TODO: think about how to reuse "gfprint" option
-                std::string filename = outputPath + "/ROMsol/romS_" + std::to_string(ti);
-                std::ofstream outfile_romS(filename.c_str());
-                outfile_romS.precision(16);
-                if (romOptions.hyperreduce && romOptions.GramSchmidt)
-                {
-                    Vector romCoord(romS);
-                    romOper[romOptions.window]->InducedGramSchmidtFinalize(romCoord, true);
-                    romCoord.Print(outfile_romS, 1);
-                }
-                else
+                if (!rom_build_database)
                 {
-                    romS.Print(outfile_romS, 1);
+                    std::string filename = outputPath + "/ROMsol/romS_" + std::to_string(ti);
+                    std::ofstream outfile_romS(filename.c_str());
+                    outfile_romS.precision(16);
+                    if (romOptions.hyperreduce && romOptions.GramSchmidt)
+                    {
+                        Vector romCoord(romS);
+                        romOper[romOptions.window]->InducedGramSchmidtFinalize(romCoord, true);
+                        romCoord.Print(outfile_romS, 1);
+                    }
+                    else
+                    {
+                        romS.Print(outfile_romS, 1);
+                    }
+                    outfile_romS.close();
                 }
-                outfile_romS.close();
 
                 if (!romOptions.hyperreduce)
-                    basis[romOptions.window]->LiftROMtoFOM(romS, *S);
+                {
+                    if (!rom_build_database || last_step)
+                    {
+                        basis[romOptions.window]->LiftROMtoFOM(romS, *S);
```

Comment:
Here, we only lift every step if we're not using rom_build_database. If we are using the greedy algorithm, then only lift at the last step (which is the 2nd to last).